### PR TITLE
Refactor: extracted transpose parameter checking routine of gemm/v

### PR DESCRIPTION
**Pull request by OldDriver233**
### What's changed?

Extracted transpose parameter checking routine into function. Currently the routines are just duplicating, resulting in bad readability.

### CODE REVIEW

File: source/module_hsolver/kernels/rocm/math_kernel_op.hip.cu [hunk 1]
```diff
@@ -653,6 +653,26 @@ void axpy_op<std::complex<double>, base_device::DEVICE_GPU>::operator()(const ba
     hipblasErrcheck(hipblasZaxpy(cublas_handle, N, (hipblasDoubleComplex*)alpha, (hipblasDoubleComplex*)X, incX, (hipblasDoubleComplex*)Y, incY));
 }
 
+hipblasOperation_t judge_trans_op(bool is_complex, const char& trans)
+{
+    if (trans == 'N')
+    {
+        return CUBLAS_OP_N;
```
Comment (dyzheng) [inline source/module_hsolver/kernels/rocm/math_kernel_op.hip.cu]: `HIPBLAS_OP_N` and same bug in code below.

**Review by dyzheng — CHANGES_REQUESTED** [review]

File: source/module_hsolver/kernels/rocm/math_kernel_op.hip.cu (hunk 1 above)
Comment (OldDriver233) [reply source/module_hsolver/kernels/rocm/math_kernel_op.hip.cu]: Fixed in commit `e54b2fe`. Also discovered bugs in `gemm_op` and fixed them.

**Review by dyzheng — APPROVED** [review]
LGTM

**Comment by haozhihan** [comment]
This PR modifies the DCU-related code, but this student probably does not have a DCU machine. So the DCU code in this PR has not been verified to be correct. 

Therefore, I am not sure whether it can be directly accepted.

**Comment by mohanchen** [comment]
LGTM

**Pull request merged**